src/libexpr/primops.cc

```diff
@@ -694,7 +694,28 @@ static void prim_genericClosure(EvalState & state, const Pos & pos, Value * * ar
 
 static RegisterPrimOp primop_genericClosure(RegisterPrimOp::Info {
     .name = "__genericClosure",
+    .args = {"attrset"},
     .arity = 1,
+    .doc = R"(
+      Take an *attrset* with elements named `startSet` and `operator` in order
+      to return an *attrset* by starting with the `startSet`, recursively
+      applying the `operator` to each element with a unique key. For example,
```

Comment:
This description doesn't mention the closure property, i.e. that it continues to call the function for each new key until it returns no new keys. Also worth mentioning that the keys must be comparable (so e.g. attrsets are not allowed).

Reply:
updated